include/deal.II/base/template_constraints.h

```diff
@@ -670,7 +670,10 @@ namespace LinearAlgebra
   {
     template <typename Number>
     class Vector;
-  }
+
+    template <typename Number, typename Node>
+    class SparseMatrix;
```

Comment:
I haven't looked at anything else yet but I think we should template on the `dealii::MemorySpace`.

Reply:
I also have to take a look into that, but if that works, let's do that!

Reply:
I tested different `dealii::MemorySpace`, but it does not compile.

I guess one needs to extend the dealii::MemorySpace first in order to work with Tpetra.

Reply:
What do you mean by that? I was mostly talking about the user-facing API. We might need to do some mapping of course.

Reply:
Oh, maybe I misunderstood your first remark.

I thought I should use `dealii::MemorySpace::(...)` as the default parameter for `Node`.

Reply:
You should be able to use `Tpetra::KokkosCompat::KokkosDeviceWrapperNode`.

Reply:
I don't quite understand the problem. This long name is just used internally
```
 template <typename Number, typename MemorySpace>
 class SparseMatrix
{
  using Node = Tpetra::KokkosCompat::KokkosDeviceWrapperNode<
    typename MemorySpace::kokkos_space::execution_space, 
    typename MemorySpace::kokkos_space>;
  ...
}
```
and everything else stays the same.

Reply:
Okay, then, let us use that one!